### DISCUSSION TITLE
C++ Scripting

### DISCUSSION THREAD

**Discussion by Ethan-Bit-256** [nem0/LumixEngine#1389] [Q&A] [answered]
Can I do Scripting in C++ purely?

**Comment by nem0** [comment, marked as answer]
Yes. What I do in my game is I make a custom c++ plugin. You can take a look at one of the existing plugins, e.g. https://github.com/nem0/LumixEngine_maps . I can post an example here later today.

**Reply by Ethan-Bit-256** [reply]
Can I do it without a plugin?

**Comment by nem0** [comment]
If you want to do it similar to Lua (or what Unity does with C#), then that's not implemented. Lumix is more like UE4 in this regard.

**Reply by Ethan-Bit-256** [reply]
Oh. In that case please do link me to this C++ plugin?

**Reply by nem0** [reply]
https://github.com/nem0/LumixEngine/wiki/Custom-plugin
